eventio/event_io_file.py

```diff
@@ -0,0 +1,65 @@
+import struct
+from .object_header import make_ObjectHeader
+from io import BytesIO
+
+class EventIOObject:
+    ''' A generic EventIOObject
+
+    It has a list of `headers` and a binary string `payload`.
+    The payload might be loaded lazily on first access or already in memory.
+    This is decided on construction time.
+    '''
+    def __init__(self, headers, payload=None, file=None):
```

Comment:
There is a class methods answer with 350 upvotes http://stackoverflow.com/a/682545/3838691 :)